src/fromager/dependencies.py

```diff
@@ -332,6 +329,51 @@ def parse_metadata(metadata_file: pathlib.Path, *, validate: bool = True) -> Met
     return Metadata.from_email(metadata_file.read_bytes(), validate=validate)
 
 
+def pep517_metadata_of_sdist(
+    *,
+    ctx: context.WorkContext,
+    req: Requirement,
+    version: Version,
+    sdist_root_dir: pathlib.Path,
+    build_env: build_environment.BuildEnvironment,
+    extra_environ: dict[str, str],
+    build_dir: pathlib.Path,
+    config_settings: dict[str, str],
+    validate: bool = True,
+) -> Metadata:
+    """Get wheel metadata from a source distribution
+
+    Uses PEP 517 prepare_metadata_for_build_wheel() API and validates
+    values.
+    """
+    hook_caller = get_build_backend_hook_caller(
+        ctx=ctx,
+        req=req,
+        build_dir=build_dir,
+        override_environ=extra_environ,
+        build_env=build_env,
+    )
+    with tempfile.TemporaryDirectory() as tmp_dir:
+        distinfo_name = hook_caller.prepare_metadata_for_build_wheel(
+            tmp_dir,
+            config_settings=config_settings,
+        )
+        metadata_file = pathlib.Path(tmp_dir) / distinfo_name / "METADATA"
+        metadata = parse_metadata(metadata_file, validate=validate)
+
+    if validate:
```

Comment:
Fetching the metadata and validation feel like different concerns that should be separate functions.

Reply:
Makes sense. I have refactored the code. Now `default_get_install_dependencies_of_sdist` is performing the validation. The location of the check is still questionable, but I cannot think of a better place without major refactoring. And I don't want to call `prepare_metadata_for_build_wheel` twice. The hook takes a non-trivial amount of time.